src/components/organisms/AssetActions/Compute/index.tsx

```diff
@@ -169,21 +169,33 @@ export default function Compute({
     ) {
       algorithmSelectionList = []
     } else {
-      const gueryResults = await queryMetadata(
+      const queryResults = await queryMetadata(
         getQuerryString(
           computeService.attributes.main.privacy.publisherTrustedAlgorithms,
           ddo.chainId
         ),
         source.token
       )
-      setDdoAlgorithmList(gueryResults.results)
-      const datasetComputeService = ddo.findServiceByType('compute')
-      algorithmSelectionList = await transformDDOToAssetSelection(
-        undefined,
-        gueryResults.results,
-        [],
-        newCancelToken()
+      // verify if the algorithms in the trusted list share the same endpoint of the dataset
+      const sameProviderEndpointAlgorithms = queryResults.results.filter(
+        (algo) =>
+          (
+            algo.findServiceByType('compute') ||
+            algo.findServiceByType('access')
+          )?.serviceEndpoint === computeService.serviceEndpoint
       )
```

Comment:
nitpick(blocking): This is 'clever' code. It uses different JavaScript features to make the code concise. But you pay with bad readability. Extract the filter into its own const instead of doing everything in one statement.

Reply:
Thought it was fine since this specific shorthand is used multiple times over the rest of the file, I'll change it to something more readable